google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/StorageObject.scala

```diff
@@ -109,7 +116,7 @@ final class StorageObject private (
 
   /** Java API */
   def withContentType(value: akka.http.javadsl.model.ContentType): StorageObject =
-    copy(contentType = value.asInstanceOf[ContentType])
+    copy(contentType = Option(value).asInstanceOf[ContentType])
```

Comment:
This looks like it would fail at runtime. The cast should be on `value`.  The `with` methods should set either both fields, `contentType` and `maybeContentType`, or just the optional field and then in the copy method do a `getOrElse` with some default value.